docs/programming.rst

```diff
@@ -1,50 +1,53 @@
 The PSI/J API
 =============
 
+The most important classes in this library are ``Job`` and ``JobExecutor``,
+followed by ``Launcher``.
 
-The PSI/J core
-~~~~~~~~~~~~~~
+The Job class and its modifiers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The core API contains the central components of PSI/J. The core API
-classes are independent of executor implementations. User code should
-only program against the core API. The major classes are:
+The Job-related classes listed in this section (``Job``, ``JobSpec``,
+``ResourceSpec``, and ``JobAttributes``) are independent of
+executor implementations. The authors strongly recommend that users
+program against these classes, rather than adding executor-specific
+configuration options, to the extent possible.
 
-:class:`~psij.job.Job`
-    Represents a scheduler job, including all its details and state.
+.. autoclass:: psij.Job
+    :members:
+    :noindex:
```

Comment:
The initial intent here was to have an overview of the classes in PSI/J so that users look at one not-so-long-page and get a rough idea of what's there. Then they can click and jump into the gory details. I'm not saying it should be one or the other, but maybe we want to give it some more thought?

Reply:
Well, I guess I agree with you if you mean a quick overview of LRMs supported. But if you open up the drop-down bar on the left it lists them all. E.g. click on PSI/J API -> Executors and you get "base class, slurm, lsf" etc. But I also think that the list could go on the front page of the project.

Reply:
Not sure what you mean. This is what I get: https://pasteboard.co/5jiFwSl41xOq.png